analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnModulus.cs

```diff
@@ -34,7 +37,15 @@ public sealed class EqualityOnModulus : SonarDiagnosticAnalyzer
         private const string DiagnosticId = "S2197";
         private const string MessageFormat = "The result of this modulus operation may not be {0}.";
 
-        private static readonly CSharpExpressionNumericConverter ExpressionNumericConverter = new CSharpExpressionNumericConverter();
+        private const string CountName = nameof(Enumerable.Count);
+        private const string LongCountName = nameof(Enumerable.LongCount);
+        private const string LengthName = "Length";
```

Comment:
Why didn't you use nameof(Array.Length); if yo used the same approach for all the other names?